RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizerForBricked.cc

```diff
@@ -0,0 +1,227 @@
+//----------------------------------------------------------------------------
+
+// Our own includes
+#include "PixelThresholdClusterizerForBricked.h"
+#include "SiPixelArrayBuffer.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelGainCalibrationOffline.h"
+// Geometry
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+//#include "Geometry/CommonTopologies/RectangularPixelTopology.h"
+
+// STL
+#include <stack>
+#include <vector>
+#include <iostream>
+#include <atomic>
+using namespace std;
+
+//----------------------------------------------------------------------------
+//! Constructor:
+//!  Initilize the buffer to hold pixels from a detector module.
+//!  This is a vector of 44k ints, stays valid all the time.
+//----------------------------------------------------------------------------
+PixelThresholdClusterizerForBricked::PixelThresholdClusterizerForBricked(edm::ParameterSet const& conf)
+    : PixelThresholdClusterizer(conf) {}
+/////////////////////////////////////////////////////////////////////////////
+PixelThresholdClusterizerForBricked::~PixelThresholdClusterizerForBricked() {}
+
+//----------------------------------------------------------------------------
+//!  \brief Cluster pixels.
+//!  This method operates on a matrix of pixels
+//!  and finds the largest contiguous cluster around
+//!  each seed pixel.
+//!  Input and output data stored in DetSet
+//----------------------------------------------------------------------------
+template <typename T>
+void PixelThresholdClusterizerForBricked::clusterizeDetUnitT(const T& input,
+                                                             const PixelGeomDetUnit* pixDet,
+                                                             const TrackerTopology* tTopo,
+                                                             const std::vector<short>& badChannels,
+                                                             edmNew::DetSetVector<SiPixelCluster>::FastFiller& output) {
+  typename T::const_iterator begin = input.begin();
+  typename T::const_iterator end = input.end();
+
+  edm::LogInfo("PixelThresholdClusterizerForBricked::clusterizeDetUnitT()");
+
+  // Do not bother for empty detectors
+  if (begin == end)
+    edm::LogWarning("clusterizeDetUnit()") << "No digis to clusterize";
+
+  //  Set up the clusterization on this DetId.
+  if (!setup(pixDet))
+    return;
+
+  theDetid = input.detId();
+
+  bool isBarrel = (DetId(theDetid).subdetId() == PixelSubdetector::PixelBarrel);
+  // Set separate cluster threshold for L1 (needed for phase1)
+  auto clusterThreshold = theClusterThreshold;
+  theLayer = (DetId(theDetid).subdetId() == 1) ? tTopo->pxbLayer(theDetid) : 0;
+  if (theLayer == 1)
+    clusterThreshold = theClusterThreshold_L1;
+
+  //  Copy PixelDigis to the buffer array; select the seed pixels
+  //  on the way, and store them in theSeeds.
+  copy_to_buffer(begin, end);
```

Comment:
please apply here same changes as in https://github.com/cms-sw/cmssw/pull/35275

Reply:
Fixed